### DISCUSSION TITLE
Issues related to error handling

### DISCUSSION THREAD

**Discussion by Hans-Wu-cn** [salvo-rs/salvo#433] [General]
this is my code:
```
#[handler]
pub async fn register(req: &mut Request,res:&mut Response) -> RebarResult<PodPool>{
    let mut parms = req.parse_json::<Pod>().await?;
    register_pod(&mut parms).await?;
    let result=get_rister_pool().await?;
    result
}

#[async_trait]
impl Writer for RebarError {
    async fn write(mut self, _req: &mut Request, depot: &mut Depot, res: &mut Response) {
        res.render(Json());
    }
}

```

I want to know how to respond to different information based on specific error types, because I don't know how to obtain the response content in the handler in the write function ,and have error in Writer trait: Only traits defined in the current crate can be implemented for arbitrary types

**Comment by chrislearn** [comment]
I don't quite understand your needs, What content you want to obtain form response?

**Comment by Hans-Wu-cn** [comment]
> I don't quite understand your needs, What content you want to obtain form response?

1. I want to handle all my exceptions uniformly. For example, if an IO exception occurs, I will use my custom exception capture and processing response JSON to the front-end
2. It seems that my handler cannot compile now. He said I did not implement salvo:: Writer, but I actually did. log: the trait bound ` Result<HashMap<std:: string:: String, Vec<Pod>, RebarError>: salvo:: Writer ` is not satisfied [E0277] Help: the trait ` salvo:: Writer ` is implemented for ` Result<T, E>`

```
#[handler]
pub async fn register(req: &mut Request,res:&mut Response) -> RebarResult<PodPool>{
    let mut parms= req.parse_json::<Pod>().await.unwrap();
    register_pod(&mut parms).await?;
    let result=get_rister_pool().await;
    result
}
```

![image](https://github.com/salvo-rs/salvo/assets/50405154/6731f18a-5db3-4882-94a6-68651e9ef4f4)

**Comment by chrislearn** [comment]
I think it is because `HashMap` is not implement `Writer`, you can wrap it use `Json`.

**Comment by Hans-Wu-cn** [comment]
> I think it is because `HashMap` is not implement `Writer`, you can wrap it use `Json`.

What does packaging use mean
Can you help me look at the second question

**Reply by chrislearn** [reply]
我不说了，你可能是`HashMap`没有实现`Writer`, 你可以 `Json` 包一下看看。
